src/main/scala/strawman/collection/immutable/ListSet.scala

```diff
@@ -124,5 +127,10 @@ object ListSet extends IterableFactory[ListSet] {
 
   def empty[A]: ListSet[A] = EmptyListSet.asInstanceOf[ListSet[A]]
 
+  def newBuilder[A](): Builder[A, ListSet[A]] =
+    new ImmutableBuilder[A, ListSet[A]](empty) {
+      def add(elem: A): this.type = { elems = elems + elem; this }
```

Comment:
I could make `ImmutableBuilder[A, C]` take as parameter a `(C, A) => C` function defining how to add an element to a given collection. Then at use site that would look like the following:

~~~ scala
new ImmutableBuilder[A, ListSet[A]](empty, _ + _)
~~~

WDYT?